### DISCUSSION TITLE
Code "protection"

### DISCUSSION THREAD

**Discussion by Filoz** [neutralinojs/neutralinojs#880] [Q&A]
Hi,
We are evaluating which framework to use for create a desktop app. A requirement we have is that the source code "protection" (no one should see the original source code).

We have found this:
 [shalithasuranga](https://github.com/shalithasuranga) commented [on 12 Apr 2021](https://github.com/neutralinojs/neutralinojs/issues/153#issuecomment-817842757) • 
Now we are using the .asar format (But it is .neu in our scenario). Therefore, I will close this issue. Thanks for reporting the issue/idea.

But, as I know, the sourcecode (js) of an asar file can be obtained.
Is there a way to make impossible to get the "original" code from the build?

Thank you!

**Comment by pegvin** [comment]
No it is not possible, your code can be obtained one way or the other, even C/C++ based application's source code can be easily obtained, you can do your best by using a JavaScript Minifier or Obfusticator.

Note the Asar format is used by Electron too, and infact you can get the source code of that application easily.

**Reply by Filoz** [reply]
Hi,
thank you for your reply!
The only solution I've found so far is to use nw.js source code protection. https://docs.nwjs.io/en/latest/For%20Users/Advanced/Protect%20JavaScript%20Source%20Code/

**Reply by pegvin** [reply]
So Source Code Protection is nothing but v8 snapshots, it's original purpose wasn't to hide the source code & infact you can decompile that `.bin` file if you have enough knowledge.

>The background for this situation is that the snapshot files are intended to be a kind of on-disk cache to speed up startup. They are not intended to be distributable binaries. ~ v8 Developer

Not taking the fact here that NeutralinoJs Doesn't come with a v8 runtime or something like that, so this is not possible.
